src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -303,15 +304,6 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys, bool autoConve
 		}
 	}
 
-	ObjectAddress tableAddress = { 0 };
-	ObjectAddressSet(tableAddress, RelationRelationId, relationId);
-
-	/*
-	 * Ensure dependencies first as we will create shell table on the other nodes
-	 * in the MX case.
-	 */
-	EnsureDependenciesExistOnAllNodes(&tableAddress);
```

Comment:
In fact, I find it confusing to have `EnsureDependenciesExistOnAllNodes` at the top, even though we don't need those dependencies till creating shell tables or shards. If we move that to the top, we also need to update `EnsureDistributedSequencesHaveOneType` as it requires to have table metadata. So, I'm inclined to keep it as it is.

Reply:
as we discussed in person, lets move above. The main reasoning is that `EnsureDependenciesExistOnAllNodes ` starts&commits a superuser initiated remote transaction, and it is sane to keep it at the top so that it never interferes with any other remote transaction. (e.g., potential deadlocks)